Documentation/cli-installation-scenarios.md

```diff
@@ -23,26 +26,99 @@ Obtaining .NET CLI
+
+From the table, we can see that if you install the SDK using `apt-get` for instance, you will get also a redist and a muxer. They will be separate packages, but will be declared as dependencies (similar for `yum`). Similar for the redist package. 
+
+The muxer is slightly a special case. Though there will be an installer, as mentioned in the previous section, it is an implementation detail. That means that acquiring the muxer should be done through either the SDK or the shared runtime installers. The only situation where this rule would not be true is if there was a major servicing event (e.g. a security update); in that case, the users would use the installer for the muxer directly, as we would rev its version accordingly. 
+
+The script installers are slightly different as they operate on zips/tarballs. The zip/tarball for the SDK will contain the entire set of things needed to be put on the disk.
+
+## Removal process
+Removing the bits from the machine **must** follow the order outlined above in installation. If the SDK is installed, it needs to be removed first and then the Redist and only then the muxer. Similar for the Redist.  
+
+## Upgrading the CLI 
+The semantics of installing the CLI will be side-by-side by default. This means that each new version will get installed besides any existing versions on disk. The [layout section](#layout-on-disk) covers how that will look like on disk.
+
+Since this is the case, there is no special "upgrade". When the user needs a new  version, the user just installs the new version using any of the installers specified in this document. The installer will just drop a new version at the predefined location. 
+
+This, however, does have one specific constraint: **newever versions must be installed in the same location the previous version was in**. This constraint is due to the fact that the "muxer" uses convention to figure out how to find the actual driver that the user reuqested. 
+
+## Layout on disk
+```
+~/dotnet
+	-dotnet (%PATH%)
+	-SDK
+       - 1.0.0
+           - 
+       - 2.0.0
+	-Shared
+		-[redist framework]
+			-1.0.0
+				-coreclr.dll
+				-mscorlib.ni.dll <Regular NI image>
+				-system.*.dll  <R2R NI image>
+				-dotnet-hostimpl.dll
+				-dotnet
+				-netcoreapp.deps 
+			-1.5.0
+				-coreclr.dll
+				-mscorlib.ni.dll <Regular NI image>-
+				-system.*.dll  <R2R NI image>
+				-dotnet-hostimpl.dll
+				-dotnet
+				-netcoreapp.deps 
+```	
+
+## Native dependencies
+.NET Core CLI is built on top of CoreFX and CoreCLR and as such its' dependencies set is defined by the platform that those two combine. Whether or not those dependencies will be installed depends on the installer being used. On Debian, for instance, using `apt-get` will mean that the appropriate dependencies are installed. For OS X using the PKG (installer) dependencies that are not part of OS X will not be installed. So, to summarize: the CLI bundle will not carry native dependencies of CoreFX and CoreCLR with it. 
```

Comment:
Fairly confusing sentence, might be more worthwhile to provide a table of which installers bring in dependencies.